### PR TITLE
Accept a boolean for the `removeQueryParameters` option

### LINKED ISSUE #133: Strip query parameters / search
Hello!

Thank you for this great library!

However, it would be convenient to also be able to strip query parameters (search).

### PR DESCRIPTION
**Pull request by SimonJang**
I saw issue #133 as unresolved so I went ahead and gave it a go. 

- Add support for `removeQueryParameters` boolean flag and implemented the new behavior
- Updated the documentation and TS types
- Added a couple of new tests.

Any feedback is welcome.

Fixes #133

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -187,6 +187,10 @@ const normalizeUrl = (urlString, options) => {
 	// Take advantage of many of the Node `url` normalizations
 	urlString = urlObj.toString();
 
+	if (options.removeQueryParameters === true) {
+		urlString = urlString.split('?')[0];
```
Comment (sindresorhus) [inline index.js:191]: This is not safe. Instead, modify the URL object.

File: readme.md [hunk 2]
```diff
@@ -187,6 +187,24 @@ normalizeUrl('www.sindresorhus.com?foo=bar&ref=test_ref', {
...
+```
+
+`false` will not remove any query parameter. Other options like `sortQueryParameters` are still applied when toggled.
+
+```js
+normalizeUrl('www.sindresorhus.com?foo=bar&utm_medium=test&ref=test_ref', {
+	removeQueryParameters: false
+});
+//=> 'http://www.sindresorhus.com/?foo=bar&ref=test_ref&utm_medium=test'
+```
```
Comment (sindresorhus) [inline readme.md:206]: index.d.ts and readme should be fully in sync.

File: readme.md [hunk 3]
```diff
@@ -187,6 +187,24 @@ normalizeUrl('www.sindresorhus.com?foo=bar&ref=test_ref', {
...
+If a boolean is provided, `true` will remove all the query parameters.
+
+```js
+normalizeUrl('www.sindresorhus.com?foo=bar', {
+	removeQueryParameters: true
+});
+//=> 'http://sindresorhus.com'
+```
+
+`false` will not remove any query parameter. Other options like `sortQueryParameters` are still applied when toggled.
```
Comment (sindresorhus) [inline readme.md:199]: I'm not sure I see the point of:

> Other options like `sortQueryParameters` are still applied when toggled.

That should already be obvious.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

File: readme.md (hunk 3 above)
Comment (SimonJang) [reply readme.md]: Removed it.

**Comment by SimonJang** [comment]
PR has been updated!

File: index.js [hunk 4]
```diff
@@ -173,6 +173,12 @@ const normalizeUrl = (urlString, options) => {
 		}
 	}
 
+	if (options.removeQueryParameters === true) {
+		for (const key of [...urlObj.searchParams.keys()]) {
```
Comment (sindresorhus) [inline index.js:177]: You can just set `urlObj.search = '';`.

**Comment by sindresorhus** [comment]
> Thanks for the feedback. PR has been updated!

Everything is not resolved.

File: index.js (hunk 4 above)
Comment (SimonJang) [reply index.js]: Nice, I didn't knew that. Changed it.

**Comment by SimonJang** [comment]
> > Thanks for the feedback. PR has been updated!
> 
> Everything is not resolved.

Should all be resolved now.

**Comment by sindresorhus** [comment]
You still need to do https://github.com/sindresorhus/normalize-url/pull/136/files#r654795467

**Comment by SimonJang** [comment]
> You still need to do [#136 (files)](https://github.com/sindresorhus/normalize-url/pull/136/files#r654795467)

Should be fully in sync now.

**Pull request merged**